### ISSUE TITLE
How to test with a custom DNN

### ISSUE THREAD

**Issue by A-Ayerh** [sitar-lab/NeuSight#1]
Hello, I first want to start off by letting you know that I appreciate your work a ton and have chosen to try to improve the latency predictor you talk about in your paper as a class project!

I've been having some issues however with setting up my custom DNN to have its latency predicted.

I wanted to start with using a simple MLP (3 hidden layers) as I made in my class and wanted to see how well it could predict the latency on an A100 series GPU.

BUT

I run into an error with setting up my model in the: "--model_config_path ../asplos/data/DLmodel_configs/" argument because of my model not matching an architecture name expected by the parser.

Could you kindly help me with figuring how to setup my model.
Thank you!

--- Heres my model:

class SimpleMLP1(nn.Module):
    def init(self):
        super(SimpleMLP1, self).init()
        self.layers = nn.Sequential(
            nn.Linear(784, 128),  # Number of input features in MNIST
            nn.ReLU(),
            nn.Linear(128, 256),
            nn.ReLU(),
            nn.Linear(256, 512),
            nn.ReLU(),
            nn.Linear(512, 128),
            nn.ReLU(),
            nn.Linear(128, 10),
            nn.Softmax()
        )

    def forward(self, x):
      x = x.view(x.shape[0], -1)  # Flatten input
      return self.layers(x)

**Comment by seonholee08** [comment]
You can customize NeuSight’s tracing to support models beyond transformers. "torch.fx", which traces PyTorch operator graphs, allows parsing of any PyTorch model. Modify the following methods to extend NeuSight’s compatibility:

- get_model and trace_fx_graph in [trace.py](https://github.com/sitar-lab/NeuSight/blob/5f123b65ea541a919f0ee06128813f4d9e28897b/neusight/Tracing/trace.py#L115) 

- parse_ops in [parse.py](https://github.com/sitar-lab/NeuSight/blob/5f123b65ea541a919f0ee06128813f4d9e28897b/neusight/Tracing/parse.py#L19) to handle unsupported operator names.
